Scripts/Diagnostics/Azure_Subscription_DP_Avoid_Plaintext_Secrets_Deployments.ps1

```diff
@@ -0,0 +1,209 @@
+    .OUTPUTS
+    [PSDeploymentOperation](https://learn.microsoft.com/dotnet/api/microsoft.azure.commands.resourcemanager.cmdlets.sdkmodels.psdeploymentoperation)
+    .EXAMPLE
+    PS> Get-ResourceGroupDeploymentOperations -SubscriptionId "00000000-xxxx-0000-xxxx-000000000000" -ResourceGroupName "MyResourceGroupName" -DeploymentName "MyDeploymentName"
+    .LINK
+    None
+  #>
+
+  [CmdletBinding()]
+  param (
+    [Parameter(Mandatory = $true)]
+    [string]
+    $SubscriptionId,
+    [Parameter(Mandatory = $true)]
+    [string]
+    $ResourceGroupName,
+    [Parameter(Mandatory = $true)]
+    [string]
+    $DeploymentName
+  )
+
+  $profile = Set-AzContext -Subscription $SubscriptionId
+
+  Get-AzResourceGroupDeploymentOperation -ResourceGroupName $ResourceGroupName -Name $DeploymentName
+}
+
+function Get-ResourceGroupDeploymentsAndOperations() {
+  <#
+    .SYNOPSIS
+    This command lists ARM deployments and operations for the Resource Group.
+    .DESCRIPTION
+    This command lists ARM deployments and operations for the Resource Group.
+    .PARAMETER SubscriptionId
+    The Azure subscription ID.
+    .PARAMETER ResourceGroupName
+    The Resource Group to which Deployments were run.
+    .INPUTS
+    None
+    .OUTPUTS
+    For each deployment: text with each deployment name, and [PSDeploymentOperation](https://learn.microsoft.com/dotnet/api/microsoft.azure.commands.resourcemanager.cmdlets.sdkmodels.psdeploymentoperation)
+    .EXAMPLE
+    PS> Get-ResourceGroupDeploymentsAndOperations -SubscriptionId "00000000-xxxx-0000-xxxx-000000000000" -ResourceGroupName "MyResourceGroupName"
+    .LINK
+    None
+  #>
+
+  [CmdletBinding()]
+  param (
+    [Parameter(Mandatory = $true)]
+    [string]
+    $SubscriptionId,
+    [Parameter(Mandatory = $true)]
+    [string]
+    $ResourceGroupName
+  )
+
+  $profile = Set-AzContext -Subscription $SubscriptionId
+
+  $deployments = Get-AzResourceGroupDeployment -ResourceGroupName $ResourceGroupName
+
+  foreach ($deployment in $deployments) {
+    Write-Debug -Debug:$debug -Message ("Deployment: " + $deployment.DeploymentName)
+
+    Get-AzResourceGroupDeploymentOperation -ResourceGroupName $ResourceGroupName -Name $deployment.DeploymentName
+  }
+}
+
+function Get-SubscriptionDeployments() {
+  <#
+    .SYNOPSIS
+    This command lists ARM deployments for the Subscription.
+    .DESCRIPTION
+    This command lists ARM deployments for the Subscription.
+    .PARAMETER SubscriptionId
+    The Azure subscription ID.
+    .INPUTS
+    None
+    .OUTPUTS
+    [PSDeployment](https://learn.microsoft.com/dotnet/api/microsoft.azure.commands.resourcemanager.cmdlets.sdkmodels.psdeployment)
+    .EXAMPLE
+    PS> Get-SubscriptionDeployments -SubscriptionId "00000000-xxxx-0000-xxxx-000000000000"
+    .LINK
+    None
+  #>
+
+  [CmdletBinding()]
+  param (
+    [Parameter(Mandatory = $true)]
+    [string]
+    $SubscriptionId
+  )
+
+  $profile = Set-AzContext -Subscription $SubscriptionId
+
+  Get-AzDeployment
```

Comment:
All these Get-Deployment output will be too verbose on console, not sure how it will help end users.

Reply:
Removed file